packages/org.json_schema.contrib/internal/TypesGenerator.pkl

```diff
@@ -63,6 +63,9 @@ function generateTypeNode(
   else if (typeNames.containsKey(schema))
     let (type = typeNames[schema as JsonSchema])
       utils.declaredType1(type, type.moduleName != enclosingModuleName)
+  else if (typeNames.containsKey((schema) { type = null }))
+    let (type = typeNames[(schema) { type = null } as JsonSchema])
+      utils.declaredType1(type, type.moduleName != enclosingModuleName)
```

Comment:
Ran into a fun case with input that looks like this:
```json
{
  "type": "object",
  "$ref": "#/$defs/users_groups.user"
}
```

Where the `schema` in the lookup `typeNames.containsKey(schema)` has `type = "object"` but the actual dereferenced schema has `type = null`. This causes a fall through to `Any` and pretty unhelpful code generation!

This particular fix is a workaround for what seems to be a common use case for `$ref`, but this doesn't handle the general case of arbitrary schema fields being set on the referrer side. I'd be very happy to take suggestions on how to handle this generally.